### DISCUSSION TITLE
Is there a way to create a JWE based on public/private keys pair, created based on strings from .env file?

### DISCUSSION THREAD

**Discussion by Hleb-Haiduk-Software** [panva/jose#456] [Q&A]
Hello Community!
I want to sign a JWE using a private key, then I want to send a public key to the Client to be able to decrypt that JWE on the Client side and read the data of the JWE.
For that, I need a private/public key pair. Also, I need a way to generate those keys pair in node.js based on the secret string I have in .env file on my backend side.
Do I follow the correct way or something is wrong with that plan?
Could not find a required example.

Thanks for the help!

**Comment by panva** [comment]
What you're describing is not at all how JOSE/JWE works.

1) there is no way to generate key pairs based off of a secret string, at most you can use KDF algorithms to derive another secrets off it, but not asymmetric key pairs.

2)

- [Algorithm Selection Guide](https://github.com/panva/jose/issues/114)
- [Algorithm Key Requirements](https://github.com/panva/jose/issues/210)

In JWE using asymmetric key management algorithms, when you're **encrypting** (not signing, JWE is encryption, not signatures) for a recipient using the recipient's **public key**. The recipient uses its private key to decrypt. This achieves confidentiality and integrity but the recipient cannot verify the JWE came from a given source because the means to encrypt is its public key.
